book/src/dev/rfcs/0010-v5-transaction.md

```diff
@@ -0,0 +1,211 @@
+
+```
+pub struct SpendV4 {
+    pub cv: commitment::ValueCommitment,
+    pub anchor: tree::Root,
+    pub nullifier: note::Nullifier,
+    pub rk: redjubjub::VerificationKeyBytes<SpendAuth>,
+    pub zkproof: Groth16Proof,
+    pub spend_auth_sig: redjubjub::Signature<SpendAuth>,
+}
+```
+
+The output code is located at `zebra-chain/src/sapling/output.rs`:
+
+```
+pub struct OutputV4 {
+    pub cv: commitment::ValueCommitment,
+    pub cm_u: jubjub::Fq,
+    pub ephemeral_key: keys::EphemeralPublicKey,
+    pub enc_ciphertext: note::EncryptedNote,
+    pub out_ciphertext: note::WrappedNoteKey,
+    pub zkproof: Groth16Proof,
+}
+```
+
+Now lets see how the v5 transaction is specified in the protocol, this is the second table of [Transaction Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus).
+
+We propose the following representation for transaction V5 in Zebra:
+
+```
+V5 {
+    lock_time: LockTime,
+    expiry_height: block::Height,
+    inputs: Vec<transparent::Input>,
+    outputs: Vec<transparent::Output>,
+    shielded_data: Option<SaplingV5ShieldedData>,
+    value_balance_sapling: Amount,
+    output_proofs_sapling: Vec<Groth16Proof>
+    anchor_sapling: tree::Root,
+    spend_zkproof_sapling: Vec<Groth16Proof>,
+    spend_auth_sigs_sapling: Vec<redjubjub::Signature<SpendAuth>>
+    orchard_data: Option<OrchardData>,
+    flags_orchard: u8,
+    value_balance_orchard: Amount,
+    anchor_orchard: tree::Root,
+}
+```
+
+`SaplingV5ShieldedData` is different from `SaplingV4ShieldedData` so a new type will be defined and implemented in `zebra-chain/src/transaction/shielded_data.rs`.
+Definition will look as follows:
+
+```
+pub struct SaplingV5ShieldedData {
+    pub first: Either<SpendV5, OutputV5>,
+    pub rest_spends: Vec<SpendV5>,
+    pub rest_outputs: Vec<OutputV5>,
+    pub binding_sig: Signature<Binding>,
+}
+```
+
+This leads to new sapling output and spend types.
+
+V5 sapling spend will be defined and implemented in `zebra-chain/src/sapling/spend.rs`(this is in the same file as `SpendV4`):
+```
+pub struct SpendV5 {
+    pub cv: commitment::ValueCommitment,
+    pub nullifier: note::Nullifier,
+    pub rk: redjubjub::VerificationKeyBytes<SpendAuth>,
+}
+```
+
+V5 sapling output will live in `zebra-chain/src/sapling/output.rs`(this is in the same file as `OutputV4`):
+
+```
+pub struct OutputV5 {
+    pub cv: commitment::ValueCommitment,
+    pub cm_u: jubjub::Fq,
+    pub ephemeral_key: keys::EphemeralPublicKey,
+    pub enc_ciphertext: note::EncryptedNote,
+    pub out_ciphertext: note::WrappedNoteKey,
+}
+```
+
+Finally, the new V5 structure will create a new `OrchardData` type. This new type will be defined in a separated file: `zebra-chain/src/transaction/orchard_data.rs` and it will look as follows:
+```
+pub struct OrchardData {
+    pub cv: commitment::ValueCommitment,
+    pub nullifier: ...,
+    pub rk: ...,
+    pub cmx: ...,
```

Comment:
Rename to `cm_x`, for consistency with Sapling's `cm_u`:
```suggestion
    pub rk: ...,
    pub cm_x: ...,
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/1886/commits/9fa5549ab218c1ba139f96384a648b2a8ba50e5f